examples/design/EU-DEMO/EUDEMO_builders/tf_coils.py

```diff
@@ -0,0 +1,805 @@
+        """
+        # The keep-out zone is for the TF WP centreline, so we need to add to it to
+        # prevent clashes when the winding pack thickness and casing are added.
+        tk_offset = 0.5 * self.params.tf_wp_width.value
+        # Variable thickness of the casing is problematic...
+        # TODO: Improve this estimate (or use variable offset here too..)
+        tk_offset += 1.5 * self.params.tk_tf_front_ib.value
+        tk_offset += self.params.g_ts_tf.value
+        return offset_wire(keep_out_zone, tk_offset, open_wire=False, join="arc")
+
+    def _derive_shape_params(self, variables_map: Dict[str, str]) -> Dict:
+        shape_params = {}
+        for key, val in variables_map.items():
+            if isinstance(val, str):
+                val = getattr(self.params, val).value
+
+            if isinstance(val, dict):
+                if isinstance(val["value"], str):
+                    val["value"] = getattr(self.params, val["value"]).value
+            else:
+                val = {"value": val}
+
+            shape_params[key] = val
+
+        # Radial width of the winding pack with no insulation or insertion gap
+        dr_wp = (
+            self.params.tf_wp_width.value
+            - 2 * self.params.tk_tf_ins.value
+            - 2 * self.params.tk_tf_insgap.value
+        )
+        # Toroidal width of the winding pack no insulation
+        dy_wp = (
+            self.params.tf_wp_depth.value
+            - 2 * self.params.tk_tf_ins.value
+            - 2 * self.params.tk_tf_insgap.value
+        )
+        # PROCESS doesn't output the radius of the current centroid on the inboard
+        r_current_in_board = (
+            self.params.r_tf_in.value
+            + self.params.tk_tf_nose.value
+            + self.params.tk_tf_ins.value
+            + self.params.tk_tf_insgap.value
+            + 0.5 * dr_wp
+        )
+
+        self.params.update_values(
+            {
+                "tk_tf_wp": dr_wp,
+                "tk_tf_wp_y": dy_wp,
+                "r_tf_current_ib": r_current_in_board,
+            },
+            source=type(self).__name__,
+        )
+
+        shape_params["x1"] = {"value": r_current_in_board, "fixed": True}
+        return shape_params
+
+    def _get_parameterisation(self) -> GeometryParameterisation:
+        return self.parameterisation_cls(self._derive_shape_params(self.variables_map))
+
+    def run(self) -> Tuple[GeometryParameterisation, BluemiraWire]:
+        """
+        Run the specified design optimisation problem to generate the TF coil winding
+        pack current centreline.
+        """
+        parameterisation = self._get_parameterisation()
+        wp_cross_section = self._make_wp_xs(self.params.r_tf_current_ib.value)
+
+        if not hasattr(self, "problem_class"):
+            raise ValueError(
+                f"Cannot execute {type(self).__name__} in RUN mode: no problem_class specified."
```

Comment:
We have a similar error on line 318, but `read` is in lowercase, where we have uppercase here. My personal preference is lower case (as that's the function name), but I'm happy with either, as long as we're consistent.